### PR TITLE
Tant caps

### PR DESCRIPTION
**Pull request by electricfungi**
Add tantalum cap functions. Fix issue with polarized cap symbol mapping

### CODE REVIEW

File: utils/db-parts.stanza [hunk 1]
```diff
@@ -788,15 +788,15 @@ defmethod to-jitx (capacitor: Capacitor) -> Instantiable :
     val esr = esr(capacitor)
     val case = case(capacitor)
     if pol? :
+      println("Got here, pol ")
```
Comment (m-hilgendorf) [inline utils/db-parts.stanza:791]: Remove print debugging

File: utils/generic-components.stanza [hunk 2]
```diff
@@ -266,6 +176,41 @@ public defn look-up-ceramic-caps (attribute: String, filter-properties:Tuple<Key
                                         filter-properties]
   look-up-capacitors(attribute, params)
 
+; ========================================================
+; ==================== tantalum-cap =======================
+; ========================================================
+
+public defn tantalum-cap (user-params:Tuple<KeyValue>) -> Instantiable :
```
Comment (m-hilgendorf) [inline utils/generic-components.stanza:183]: You can add doc string for public functions to describe the API : 

```
doc: "Create a tantalum cap. Supported key/value args are \"capacitance\", \"tolerance\", etc."
public defn tantalum-cap (user-params:Tuple<KeyValue>) -> Instantiable :

doc: "Create a tantalum cap with given capacitance"
public defn tantalum-cap (cap:Double) -> Instantiable : 

doc: "Create a tantalum cap with a given capacitance and tolerance" 
public defn tantalum-cap (cap:Double, tolerance:Double) -> Instantiable : 
```

The `doc:` syntax is there for stanza's autodocumentation feature. We should start using it more so we can auto-generate OCDB's docs.

File: utils/generic-components.stanza [hunk 3]
```diff
@@ -266,6 +176,41 @@ public defn look-up-ceramic-caps (attribute: String, filter-properties:Tuple<Key
...
+  val params = remove-duplicate-keys $ [["anode" => "tantalum" "min-rated-voltage" => MIN-CAP-VOLTAGE]
+                                        smd-query-properties()
+                                        user-params]
+  map(to-jitx, Capacitors(params, limit))
+
+; ========================================================
+; =============== look-up-tantalum-caps ===================
+; ========================================================
+
+public defn look-up-tantalum-caps (attribute: String) -> Tuple :
```
Comment (m-hilgendorf) [inline utils/generic-components.stanza:205]: What are the valid values for the `attribute` argument and what does it control? 

What is the return of this? Is it `Tuple<JSON>`?

File: utils/generic-components.stanza [hunk 4]
```diff
@@ -266,6 +176,41 @@ public defn look-up-ceramic-caps (attribute: String, filter-properties:Tuple<Key
...
+                                        user-params]
+  to-jitx $ Capacitor(params)
+
+public defn tantalum-cap (cap:Double) -> Instantiable :
+  tantalum-cap(["capacitance" => cap])
+
+public defn tantalum-cap (cap:Double, tolerance:Double) -> Instantiable :
+  tantalum-cap(["capacitance" => cap "tolerance" => tolerance])
+
+public defn tantalum-caps (user-params:Tuple<KeyValue>, limit: Int) -> Tuple<Instantiable> :
```
Comment (m-hilgendorf) [inline utils/generic-components.stanza:195]: This function is interesting, it returns a tuple of valid components that satisfy your query. How would it be used in context, compared to using `lookup-tantalum-caps' and selecting one?

**Review by m-hilgendorf — COMMENTED** [review]
We should have unit tests for these functions, something like this : 

```
#use-added-syntax(jitx, tests)
defpackage ocdb/test/tantalum-caps : 
  import core
  import collections
  import jitx
  import jitx/commands
  import ocdb/generic-components

deftest tantalum-cap-instantiation-1uf : 
  pcb-module module : 
     inst cap : tantalum-cap(1.0e-6)
     #EXPECT(property(cap.capacitance) == 1.0e-6)
     #EXPECT(property(cap.anode) == "tantalum")
   evaluate(module)
   
deftest tantalum-cap-lookup-1uf : 
  val caps = lookup-tantalum-cap(..., ["capacitance" => 1.0e-6])
  #ASSERT(not empty?(caps))
```

File: utils/db-parts.stanza (hunk 1 above)
Comment (electricfungi) [reply utils/db-parts.stanza]: fixed

File: utils/generic-components.stanza (hunk 2 above)
Comment (electricfungi) [reply utils/generic-components.stanza:183]: pass - this is a new doc flow beyond the scope of this change

File: utils/generic-components.stanza (hunk 3 above)
Comment (electricfungi) [reply utils/generic-components.stanza:205]: pass - this gets captured in our docs

File: utils/generic-components.stanza (hunk 4 above)
Comment (electricfungi) [reply utils/generic-components.stanza:195]: pass - this gets captured in our docs

File: utils/generic-components.stanza (hunk 3 above)
Comment (m-hilgendorf) [reply utils/generic-components.stanza:205]: I don't see any docs for this.

**Review by m-hilgendorf — APPROVED** [review]

**Pull request merged**